### ISSUE TITLE
[[ -z "$@" ]] should probably warn about unary -z mismatch with non-unary "$@"

### ISSUE THREAD

**Issue by floatingatoll** [koalaman/shellcheck#812]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash

if [ -z "$@" ]; then true; fi
if [[ -z "$@" ]]; then true; fi
```

#### Here's what shellcheck.net currently says:
```No issues detected!```

#### Here's what I wanted or expected to see:
```-z takes unary argument but "$@" expands to multiple arguments, use "$*" instead```

**Comment by kurahaupo** [comment]
I'm sure there are valid uses too:

set -- "" -a -d /etc
[ -z "$@" ] && echo /etc is a dir
[[ -z "$@" ]] && echo args are not empty

OK, that might be straining the point.

**Comment by floatingatoll** [comment]
That was sort of my theory, yes: while a legit use could occur, it is almost certainly going to be an edge case. Virtually all uses of that type would simply be [[ "$@" ]], right?

**Comment by koalaman** [comment]
Good idea. This has been added in 6aee12a , along with improved warnings for brace expansions and globs.

I left `[ "$@" ]` out to allow dynamically building tests.

**Comment by floatingatoll** [comment]
Agreed - the standalone "$@" case makes perfect sense. Thank you!

**Comment by floatingatoll** [comment]
#885.
